src/pybind/mgr/status/module.py

```diff
@@ -244,9 +244,10 @@ def handle_fs_status(self, fs: Optional[str] = None, format: str = 'plain'):
 
         if len(mds_versions) == 1:
             if output_format in ('json', 'json-pretty'):
-                json_output['mds_version'] = list(mds_versions)[0]
+                json_output['mds_version'] = [dict(version=k, daemon=v)
+                                              for k, v in mds_versions.items()]
```

Comment:
why do we omit the `mds_version` key in the `json_output`, if we have move than a single MDS version? Your change modifies the structure that it now works for more than a single version, but we're still in the `if len(mds_versions) == 1:` case here.

Reply:
what the existing version omits is the daemon names, not their versions.

i think the idea was to have a simpler output if all MDS daemons are of the same version. and i guess the author believed that it didn't matter what the version mapping was if we only had a single version.

in that case, before this change, `json_output['mds_version']` would be a single version string. after my change, it's consistent with the case where we have multiple versions of MDS daemons in the cluster.